### PR TITLE
[C-1195] Fix image loading in image hooks

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

- Fixes pretty interesting/bad bug where we are dispatching fetch image actions in the render cycle by moving dispatch to useEffect/useCallback depending on onDemand value)
- Updates types to properly type useImage based on onDemand value
- Updates usages, particularly onDemand cases (in ArtistPopover) to use special hook rather than overloading useProfileImage and useCoverPhotoImage to match the useImage types in common

### CODE REVIEW

File: packages/common/src/hooks/useImageSize.ts [hunk 1]
```diff
@@ -82,77 +97,121 @@ export const useImageSize = <
...
     if (larger) {
-      return fallbackImage(larger)
-    }
-
-    // If no larger size exists, dispatch to get the desired size
-    // Don't dispatch twice for the same id
-    if (load && getPreviousId() !== id) {
-      setPreviousId(id)
-      // Request the desired size
-      dispatch(action(id, size))
```
Comment (dylanjeffers) [inline packages/common/src/hooks/useImageSize.ts:137]: This was the issue, since this was technically happening in the render cycle

Comment (raymondjacobson) [reply packages/common/src/hooks/useImageSize.ts:137]: ooooooouyyydddddffff

i'm sure i wrote this

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/dj-c-1195-fix-image-loading

File: packages/common/src/hooks/useImageSize.ts [hunk 2]
```diff
@@ -82,77 +97,121 @@ export const useImageSize = <
...
-  return getImageSize as any
+  const previousId = getPreviousId()
+
+  const handleFetchLargeImage = useCallback(
+    (imageType) => {
+      if (
+        load &&
+        !(id === null || id === undefined || typeof id === 'string') &&
+        previousId !== id &&
+        (imageType === 'smaller' || imageType === 'undefined')
```
Comment (raymondjacobson) [inline packages/common/src/hooks/useImageSize.ts:189]: str undefined?

**Comment by raymondjacobson** [comment]
gonna wait for demo site and scrutinize behavior a little then approve

**Comment by raymondjacobson** [comment]
Screen Shot 2022-09-28 at 6 11 10 PM
Some staleness issue on profile photos.
Repro:
1. go to feed
2. click on profile in track tile
3. go back
4. click on profile without a picture in track tile
here's an entropy key if that's useful for a user where this happens on their feed: f0766d4f5d92555591b77b076c6cba8a

File: packages/common/src/hooks/useImageSize.ts (hunk 2 above)
Comment (raymondjacobson) [reply packages/common/src/hooks/useImageSize.ts:189]: oh i see nvm. pushing a change to change type to a string union

Comment (dylanjeffers) [reply packages/common/src/hooks/useImageSize.ts:189]: Yeah that was v quick and dirty

**Comment by raymondjacobson** [comment]
> Screen Shot 2022-09-28 at 6 11 10 PM
> 
> Some staleness issue on profile photos. Repro: 1. go to feed 2. click on profile in track tile 3. go back 4. click on profile without a picture in track tile here's an entropy key if that's useful for a user where this happens on their feed: f0766d4f5d92555591b77b076c6cba8a

OK this bug is on staging already. NVM, will open sep issue

**Review by raymondjacobson — APPROVED** [review]

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/dj-c-1195-fix-image-loading

**Comment by dylanjeffers** [comment]
Gotcha yeah was gonna say I think we saw this on stage. Kewl thanks for approval

**Pull request merged**